src/hotspot/share/cds/metaspaceShared.cpp

```diff
@@ -758,8 +758,10 @@ void MetaspaceShared::link_shared_classes(bool jcmd_request, TRAPS) {
       ClassLoaderData* cld = collect_cld.cld_at(i);
       for (Klass* klass = cld->klasses(); klass != nullptr; klass = klass->next_link()) {
```

Comment:
Why doesn't this call loaded_classes_do() ?

Reply:
I update the PR to use `ClassLoaderDataGraph::loaded_classes_do_keepalive()`, so there's no need to explicitly check for `is_loaded()`.

Also, instead of holding onto the ClassLoader oops which has a side effect of keeping the classes alive, I am now holding onto the Java mirrors directly. This simplifies the code and hopefully makes it easier to understand.